s3tests/functional/test_s3.py

```diff
@@ -5351,7 +5351,10 @@ def test_multipart_copy_invalid_range():
     bucket, key = _create_key_with_random_content('source', size=5)
     upload = bucket.initiate_multipart_upload('dest')
     e = assert_raises(boto.exception.S3ResponseError, copy_part, bucket.name, key.name, bucket, 'dest', upload.id, 0, 0, 21)
-    eq(e.status, 400)
+    # on RHEL we get response 416, check bugzilla 1501005
+    valid_status = [400, 416]
+    if not any(s == e.status for s in valid_status):
+       raise AssertionError("Invalid response " + str(status))
```

Comment:
this will accept 400 or 416 for `e.status`, but the `e.reason` and `e.error_code` checks below are still specific to `400 Bad Request` - right?

Reply:
I haven't got past due to first assert, my guess was reason and error code should be same, I will know in new runs and update here.

Reply:
ok. for 416, i would expect `e.reason` to be [Requested Range Not Satisfiable](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.17). for `e.error_code`, i don't think the requirements are well-defined, so it's probably best to ignore it

Reply:
I am bit surprised the tests worked here: http://pulpito.ceph.redhat.com/vasu-2017-11-03_14:24:45-rgw:multifs-luminous---basic-multi

Compared to one without fix:
http://pulpito.ceph.redhat.com/vasu-2017-11-01_18:16:45-rgw:multifs-luminous---basic-multi/

Reply:
huh checking the branch, it used ceph-master, :(